JavaScript/JavaScriptSDK.Tests/CheckinTests/Sender.tests.ts

```diff
@@ -655,12 +666,497 @@ class SenderTests extends TestClass {
                 Microsoft.ApplicationInsights.Util.canUseSessionStorage = utilCanUserSession;
             }
         });
+
+        this.testCase({
+            name: "SenderTests: XMLHttpRequest sender can handle partial success errors. Non-retryable",
+            test: () => {
+                // setup
+                XMLHttpRequest = <any>(() => {
+                    var xhr = new this.xhr;
+                    xhr.withCredentials = false;
+                    return xhr;
+                });
+
+                this.validatePartialSuccess_NonRetryable();
+            }
+        });
+
+        this.testCase({
+            name: "SenderTests: XMLHttpRequest sender can handle partial success errors. Retryable",
+            test: () => {
+                // setup
+                XMLHttpRequest = <any>(() => {
+                    var xhr = new this.xhr;
+                    xhr.withCredentials = false;
+                    return xhr;
+                });
+
+                this.validatePartialSuccess_Retryable();
+            }
+        });
+
+        this.testCase({
+            name: "SenderTests: XMLHttpRequest - can disable partial response handling",
+            test: () => {
+                // setup
+                XMLHttpRequest = <any>(() => {
+                    var xhr = new this.xhr;
+                    xhr.withCredentials = false;
+                    return xhr;
+                });
+
+                this.validatePartialSuccess_disabled();
+            }
+        });
+
+        this.testCase({
+            name: "SenderTests: handles retry-after value returned in the response header (XMLHttpRequest)",
+            test: () => {
+                // setup
+                XMLHttpRequest = <any>(() => {
+                    var xhr = new this.xhr;
+                    xhr.withCredentials = false;
+                    return xhr;
+                });
+
+                var sender = this.getSender();
+                Assert.ok(sender, "sender was constructed");
+
+                // send two items
+                this.fakeServer.requests.pop();
+                sender.send(this.testTelemetry);
+                sender.send(this.testTelemetry);
+
+                Assert.equal(2, sender._buffer.count(), "Buffer has two items");
+
+                // trigger send
+                this.clock.tick(sender._config.maxBatchInterval());
+
+                // retry after 777s. 
+                var now = 1468864738000;
+                this.clock.setSystemTime(now);
+
+                var delay = 77; 
+                var delayDate = new Date();
+                delayDate.setSeconds(delayDate.getSeconds() + delay);
+
+                this.requestAsserts();
+                this.fakeServer.requests.pop().respond(
+                    206,
+                    { "Content-Type": "application/json", "Retry-After": delayDate.toUTCString() },
```

Comment:
So for this test with a non-retryable 408 shouldn't the retry-after header not exist?